commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/unicycle/control/UnicycleCoreTasks.scala

```diff
@@ -385,4 +385,39 @@ trait UnicycleCoreTasks {
     }
   }
 
+  class DriveToTargetWithConstantSpeed(drivetrainComponent: Drivetrain,
+                                       distanceToTarget: Stream[Option[Length]],
+                                       angleToTarget: Stream[Angle],
+                                       forwardVelocity: Dimensionless,
+                                       maxTurnVelocity: Dimensionless,
+                                       minDistance: Length)
+                                      (implicit drivetrainHardware: DrivetrainHardware,
+                                       props: Signal[DrivetrainProperties]) extends FiniteTask {
+
+    override def onStart(): Unit = {
+      val absoluteTargetAngle = drivetrainHardware.turnPosition.zipAsync(angleToTarget).map {t =>
+        t._1 + t._2
+      }
+
+      val turnController = turnPositionControl(absoluteTargetAngle)._1
+
+      val out = childVelocityControl(
+        turnController.map { p =>
+          UnicycleSignal(forwardVelocity, p.turn min maxTurnVelocity max -maxTurnVelocity)
+        }
+      )
+
+      drivetrainComponent.setController(out.withCheckZipped(distanceToTarget){
+        distanceToTarget => {
+          if (!distanceToTarget.exists(_ > minDistance)) {
```

Comment:
Shouldn't this be a exists < minDistance? Right now this task will finish is no cube is detected.